### PR TITLE
[UI] Transaction Date Range: persistence added and format modified

### PR DESCRIPTION
**Pull request by crowning-**
Reference: https://github.com/dashpay/dash/issues/879

It's a compromise between requested functionality and overcrowded wallet-UI.

1. - the selected option (_"Range..."_)  and the chosen dates of the range are persisted now
2. - date format changed to standard date format

Looks like this now:
date_range

### CODE REVIEW

**Comment by UdjinM6** [comment]
Looks good 👍 
utACK a4403eb

**Comment by schinzelh** [comment]
utACK a4403eb

**Comment by crowning-** [comment]
I think, after a wallet reset, the default date range _"2000-01-01 / 2000-01-01"_ isn't the best default.

I'll change this to _currentDate / currentDate+1_ tomorrow.

File: src/qt/transactionview.cpp [hunk 1]
```diff
@@ -295,15 +300,16 @@ void TransactionView::chooseDate(int idx)
                 QDateTime(QDate(current.year(), 1, 1)),
                 TransactionFilterProxy::MAX_DATE);
         break;
-    case Range:
+    case Range:
```
Comment (UdjinM6) [inline src/qt/transactionview.cpp]: Nit: spaces

File: src/qt/transactionview.cpp [hunk 2]
```diff
@@ -516,24 +527,26 @@ QWidget *TransactionView::createDateRangeWidget()
...
+    // Load persisted TO date
+    dateTo->setDate(QDate::fromString(settings.value("transactionDateTo", defaultDateTo).toString(), PERSISTENCE_DATE_FORMAT));
+
     layout->addWidget(dateTo);
     layout->addStretch();
 
     // Hide by default
     dateRangeWidget->setVisible(false);
-
+
```
Comment (UdjinM6) [inline src/qt/transactionview.cpp]: Nit: spaces

**Comment by crowning-** [comment]
Default value set to _currentDate / currentDate+1_, minor refactoring, squashed into 1 commit.

Edit: spaces? Which spaces?

**Comment by UdjinM6** [comment]
heh, ok...
Permissions? 😉 
> transactionview.cpp 100644 → 100755

**Comment by crowning-** [comment]
Aargh, I shouldn't use Windows programs to edit stuff...

Reverted permissions and re-squashed.

**Comment by UdjinM6** [comment]
Looks good again now :)
utACK b7546d2

**Pull request merged**